pkg/eestream/decode.go

```diff
@@ -285,31 +285,36 @@ func (dr *decodedRanger) Range(offset, length int64) io.ReadCloser {
 	// do it parallel to save from network latency
 	readers := make(map[int]io.ReadCloser, len(dr.rrs))
 	type indexReadCloser struct {
-		i int
-		r io.ReadCloser
+		i   int
+		r   io.ReadCloser
+		err error
 	}
 	result := make(chan indexReadCloser, len(dr.rrs))
 	for i, rr := range dr.rrs {
 		go func(i int, rr ranger.Ranger) {
-			r := rr.Range(
+			r, err := rr.Range(
 				firstBlock*int64(dr.es.EncodedBlockSize()),
 				blockCount*int64(dr.es.EncodedBlockSize()))
-			result <- indexReadCloser{i, r}
+			result <- indexReadCloser{i, r, err}
```

Comment:
while you're here, better to name your struct fields

Reply:
Yes, of course.